tfconfig/modules/vm-instance-group/main.tf

```diff
@@ -88,15 +107,38 @@ resource "google_compute_instance_template" "compute_vm_template" {
     auto_delete = true
   }
 
-  network_interface {
-    dynamic "access_config" {
-      for_each = var.disable_public_ips == true ? [] : [1]
-      content {}
-    }
+  dynamic "network_interface" {
+    for_each = local.network_interfaces
 
-    network    = var.network_self_link
-    subnetwork = var.subnetwork_self_link
-    nic_type   = local.enable_gvnic ? "GVNIC" : null
+    content {
```

Comment:
Maybe indent to make it clear that it's part of `for_each`

Reply:
Seems like this is the style convention for terraform. https://developer.hashicorp.com/terraform/language/expressions/dynamic-blocks